Source/DafnyCore/Compilers/SinglePassCompiler.cs

```diff
@@ -64,6 +65,39 @@ public abstract class SinglePassCompiler : Plugins.Compiler {
 
     public CoverageInstrumenter Coverage;
 
+    public ProcessStartInfo PrepareProcessStartInfo(string programName, IEnumerable<string> args = null) {
+      var psi = new ProcessStartInfo(programName) {
+        UseShellExecute = false,
+        CreateNoWindow = false, // https://github.com/dotnet/runtime/issues/68259
+      };
+      foreach (var arg in args ?? Enumerable.Empty<string>()) {
+        psi.ArgumentList.Add(arg);
+      }
+      return psi;
+    }
+
+    public int WaitForExit(Process process, TextWriter outputWriter, string errorMessage = null) {
+      process.WaitForExit();
+      if (process.ExitCode != 0 && errorMessage != null) {
+        outputWriter.WriteLine("{0} Process exited with exit code {1}", errorMessage, process.ExitCode);
+      }
+      return process.ExitCode;
+    }
+
+    public int RunProcess(ProcessStartInfo psi, TextWriter outputWriter, string errorMessage = null) {
+      try {
+        var process = Process.Start(psi);
+        if (process == null) {
+          return -1;
+        }
+        return WaitForExit(process, outputWriter, errorMessage);
+      } catch (System.ComponentModel.Win32Exception e) {
```

Comment:
What happens if the process fails to start on a non-Windows machine, does it just return -1 then? Shouldn't the branch on line 91 and on line 95 have similar behavior?

Reply:
No, it throws a Win32 exception on all platforms, which is a bit odd (at least that's what it's documented to do).

Reply:
System.ComponentModel.Win32Exception